packages/clarity-js/src/layout/region.ts

```diff
@@ -1,108 +1,77 @@
-import { Event, BooleanFlag } from "@clarity-types/data";
-import { Box, RegionData } from "@clarity-types/layout";
+import { BooleanFlag, Event } from "@clarity-types/data";
+import { RegionData } from "@clarity-types/layout";
+import * as dom from "@src/layout/dom";
 import encode from "@src/layout/encode";
-import * as dom from "./dom";
 
-let bm: { [key: number]: RegionData } = {};
-let updateMap: number[] = [];
+export let data: RegionData[] = [];
+let regionMap: WeakMap<Node, string> = null; // Maps region nodes => region name
+let regions: { [key: number]: RegionData } = {};
+let updates: number[] = [];
+let watch = false;
+let observer: IntersectionObserver = null;
 
-export function compute(): void {
-    // Making region compute synchronous for tracking regions 
-    // TODO: For modern browsers use resizeobserver 
-    region();
-}
-
-async function region(): Promise<void> {
-    let values = dom.regions();
-    let viewport = getViewport();
-
-    for (let value of values) {
-        update(value.id, layout(dom.getNode(value.id) as Element, viewport));
-    }
-
-    if (updateMap.length > 0) { await encode(Event.Region); }
+export function start(): void {
+    reset();
+    observer = null;
+    regionMap = new WeakMap();
+    regions = {};
+    watch = window["IntersectionObserver"] ? true : false;
```

Comment:
Will this lower region coverage based on intersection observer support in browsers?

Reply:
No real impact since this API has been around for a while: https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API#browser_compatibility